src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -1182,6 +1208,59 @@ export class StudyViewPageStore
                             this.molecularProfileMapByType,
                             selectedSamples
                         );
+                    } else if (chartMeta.dataType === 'Custom_Data') {
+                        const clinicalAttribute = chartMeta.clinicalAttribute;
+                        const sampleKeyToData = this._customChartsSelectedCases.get(
+                            chartMeta.uniqueKey
+                        );
+                        if (clinicalAttribute) {
+                            if (selectedSamples.length === 0) {
+                                data = [];
+                            } else {
+                                if (sampleKeyToData !== undefined) {
+                                    const sampleInformation = sampleKeyToData.map(
+                                        item => ({
+                                            sampleId: item.sampleId,
+                                            studyId: item.studyId,
+                                            patientId: item.patientId,
+                                            value: item.value,
+                                        })
+                                    );
+                                    // Combine selectedSamples with sampleKeyToData to get the desired data object
+                                    let merged = [];
+                                    for (
+                                        let i = 0;
+                                        i < selectedSamples.length;
+                                        i++
+                                    ) {
+                                        merged.push({
+                                            ...selectedSamples[i],
+                                            ...sampleInformation.find(
+                                                itmInner =>
+                                                    itmInner.sampleId ===
+                                                    selectedSamples[i].sampleId
+                                            ),
+                                        });
+                                    }
+
+                                    const inputObject: any[] = merged.filter(
```

Comment:
why would uniqueSampleKey ever be undefined?

if there is a valid reason, this filter operation can be chained with the map above.  

```
const mergedWhatevers = arr.map(...)
    .filter(...);
```

```

Reply:
The `uniqueSampleKey` is undefined in the `selectedSampleData` when the sample is not in `selectedSamples`. When we filter based on the `uniqueSampleKey`, we only send the selected samples in the studyview filter to the group comparison. (I think an inner join would have been better)